lib/matplotlib/image.py

```diff
@@ -1190,6 +1189,19 @@ def set_cmap(self, cmap):
             raise RuntimeError('Cannot change colors after loading data')
         super().set_cmap(cmap)
 
+    def get_cursor_data(self, event):
+        # docstring inherited
+        x, y = event.xdata, event.ydata
+        if (x < self._Ax[0] or x > self._Ax[-1] or
+                y < self._Ay[0] or y > self._Ay[-1]):
+            return None
+        j = np.searchsorted(self._Ax, x) - 1
+        i = np.searchsorted(self._Ay, y) - 1
+        try:
+            return self._A[i, j]
+        except IndexError:
```

Comment:
Can this case actually occur?  (I guess(?) perhaps Ax/Ay could be out of sync in size with A, but in that case there's no reason to believe that the result would make sense if the indexing succeeds either.)

Reply:
Good question, I think this was lifted from `PcolorImage` which doesn't seem to be tested... I guess it doesn't hurt to have this safeguard here?

Reply:
My personal PoV is that we should not introduce random try... except... guards unless we have a good reason to believe that they are needed; OTOH I'm not going to block this over that.

Reply:
I agree that when I read this I was wondering why we would hit an exception here because you're handling the edge cases above. I'd remove the try/except here, especially because this is in a hot-loop with mouse movements.

Reply:
👍 removed. I also removed it from the `PcolorImage` implementation that this was originally copied from. Looking at the original PR (https://github.com/matplotlib/matplotlib/pull/6930) I see no specific reason it was added, and it is untested in `PcolorImage`.